### PR TITLE
Lambda explicit return type: IDE tests

### PR DESCRIPTION
**Pull request by cston**
Proposal: [lambda-improvements.md](https://github.com/dotnet/csharplang/blob/main/proposals/lambda-improvements.md#explicit-return-type)
Test plan: https://github.com/dotnet/roslyn/issues/52192

### CODE REVIEW

File: src/Workspaces/CSharpTest/Formatting/FormattingTests.cs [hunk 1]
```diff
@@ -9973,5 +9973,61 @@ public async Task RecordStruct()
...
+        public async Task LambdaReturnType_01()
+        {
+            await AssertFormatAsync(
+@"class Program
+{
+    Delegate D = void () => { };
+}",
+@"class Program
+{
+    Delegate D = void  ()  =>  {  };
```
Comment (CyrusNajmabadi) [inline src/Workspaces/CSharpTest/Formatting/FormattingTests.cs:9987]: wacky :)

Comment (CyrusNajmabadi) [reply src/Workspaces/CSharpTest/Formatting/FormattingTests.cs:9987]: (oh wait, is this one of those weir dtest harnesses that puts the 'expected' first?

**Review by CyrusNajmabadi — APPROVED** [review]

File: src/Workspaces/CSharpTest/Formatting/FormattingTests.cs (hunk 1 above)
Comment (cston) [reply src/Workspaces/CSharpTest/Formatting/FormattingTests.cs:9987]: Yes, `expected` is the first parameter.

**Comment by cston** [comment]
Rebased https://github.com/dotnet/roslyn/commit/a1f5815d2fa7cc52e223bd904131a31814a643ab after merging https://github.com/dotnet/roslyn/pull/54075.

**Pull request merged**